red/reporter.js

```diff
@@ -0,0 +1,54 @@
+/**
+ * Copyright 2015 IBM Corp.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ **/
+
+var util = require('util');
+var EventEmitter = require('events').EventEmitter;
+var Log = require("./log");
+var TEN_SECONDS_MILLIS = 10000;
+
+var reporter = {
+    init: function(settings) {
+        if(settings) {
+            // TODO settings might be used in the future for configuring runtime reports
+        }
+        startInterval(TEN_SECONDS_MILLIS);
+    }
```

Comment:
I don't understand why this is needed? Surely we just either want to report memory metrics or not, that could be controlled from the settings file.

Are you suggesting this because of the possibility of future features/what ifs in which case it's best if this is already tied in (as we'd likely to forget to add a hook for memory reporter's shutdown when adding such feature)?